Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppDeploy/MsDeployOnTargetMachines.ps1

```diff
@@ -77,7 +81,25 @@ function Get-MsDeployCmdArgs
         throw "Package does not exist : `"$webDeployPackage`""
     }
 
-    $msDeployCmdArgs = [string]::Empty
+    $msDeployCmdArgs = [String]::Format(" -verb:sync")
+
+    if($isFolderBasedDeployment)
+    {
+        $msDeployCmdArgs += [String]::Format(' -source:iisApp="{0}"', $webDeployPackage);
+        $msDeployCmdArgs += [String]::Format(' -dest:iisApp="{0}"', $websiteName);
+    }
+    else
+    {
+        $msDeployCmdArgs += [String]::Format(' -source:package="{0}"', $webDeployPackage);
+        if($isParamFilePresentInPacakge)
+        {
+            $msDeployCmdArgs += [String]::Format(' -dest:auto');
+        }
+        else {
+            $msDeployCmdArgs += [String]::Format(' -dest:contentPath="{0}",', $websiteName);
```

Comment:
In above mentioned issue, since package is being generated without mentioning providerPath, while deploying package WebDeploy try to deploy package at  sourcePath(O:\Test) from which package is being generated. So issue is with package generation. Package in right way could be generated using below command 
`"C:\Program Files\IIS\Microsoft Web Deploy V3\msdeploy.exe" -source:iisApp="O:\TestSite" -dest:package="WebApp.zip" -declareParam:name="IIS Web Application Name",defaultValue="WebApp",tags=IisApp,scope=IisApp,kind=ProviderPath  -declareParam:name="IIS Web Application Name",scope=setAcl,kind=ProviderPath -replace:match="O:\TestSite",replace="WebApp" -verb:sync -retryAttempts:20 -disableRule:BackupRule`